src/helm/proxy/clients/huggingface_model_registry.py

```diff
@@ -71,36 +76,54 @@ def from_string(raw: str) -> "HuggingFaceModelConfig":
             namespace=match.group("namespace"), model_name=model_name, revision=match.group("revision")
         )
 
+    @staticmethod
+    def from_path(path: str) -> "HuggingFaceModelConfig":
+        """Genertes a HuggingFaceModelConfig from a (relative or absolute) path to a local HuggingFace model."""
+        pattern = r"((?P<containing_dir>.+?)/)?(?P<model_name>[^/@]+)(@(?P<revision>[^/@]+))?/?"
+        match = re.fullmatch(pattern, path)
+        if not match:
+            raise ValueError(f"Could not parse model path: '{path}'; Expected format: /path/to/model_name[@revision]")
+        model_name = match.group("model_name")
+        assert model_name
+        return HuggingFaceModelConfig(
+            namespace="local",
+            model_name=model_name,
+            revision=match.group("revision"),
+            path=path
+        )
 
 _huggingface_model_registry: Dict[str, HuggingFaceModelConfig] = {}
```

Comment:
Let's just pre-register our local Hugging Face models in `_huggingface_model_registry`:

```
{
    "huggingface/llama-7b" : HuggingFaceModelConfig(
        namespace="huggingface",
        model_name="llama-7b",
        revision=None,
        path=os.path.join("${LOCAL_HUGGING_FACE_MODEL_DIR", "llama-7b"),
}
```

Then we can get rid of the `local` group.

Eventually we should also move the special-cased things like `EleutherAI/gpt-j-6B` here as well, but not in this PR.

Reply:
this does make things easier in some ways, but we just have to remember to duplicate any local models both in `models.py` and here

Reply:
just implemented this -- the code is much cleaner now!